[MM-20533] Migrate "RhsHeaderPost" component to TypeScript

utkuufuk opened the pull request:
#### Summary
Migrates `components/rhs_header_post` module to TypesScript.

#### Ticket Link
Jira: https://mattermost.atlassian.net/browse/MM-20533
Fixes https://github.com/mattermost/mattermost-server/issues/13708

devinbinnie commented on components/rhs_header_post/rhs_header_post.tsx:
```diff
@@ -12,27 +11,25 @@ import Constants, {RHSStates} from 'utils/constants';
+    relativeTeamUrl: string;
+    channel: {display_name: string};
+    actions: {
+        setRhsExpanded: Function;
```
Can we be more specific as to what the functions are here? Otherwise we could technically pass in any function as a prop here, and it kinda circumvents type checking.

utkuufuk commented:
> Looks good overall, just a couple changes to make.
> Thanks very much for your PR @utkuufuk :)

@devinbinnie I've addressed your comments, thanks for reviewing 🙂
